### PR TITLE
Editor: Debounce autosave on last edit

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to change the behavior of autosave such that an auto save does not occur until after the delay from the last edit made. Thus, so long as a user continues typing, an autosave will never occur until they are finished typing.

**Testing instructions:**

Verify that upon making a change to e.g. the title or content, an autosave occurs after 10 seconds of no action. Verify the implications of this specific behavior in that repeatedly making changes before the 10 seconds interval will reset the timer.

### CODE REVIEW

**Comment by tecnogaming** [comment]
We need to have more control on the autosave feature.  You need to understand that for tiny blogs there is no problem with the current implementation of autosave but for big blogs with up to 9 editors/writers all working at the same time, the autosave and the amount it does is overkill.  

We DONT NEED that much autosave, we need to limit or disable it and there is no option.  Gutenberg should not decide for itself,  this needs to be added as an option to the user.

**Comment by aduth** [comment]
I forgot I'd left this in a half-finished state. This should now be ready for review.

Since the initial commit, I've revised the approach to avoid relying on a non-deterministic value being introduced to the reducer state, instead taking an approach not too much unlike what we have with [`getBlockDependantsCacheBust`](https://github.com/WordPress/gutenberg/blob/29f209a767937e3497f12296ecab0f2de09f592f/packages/editor/src/store/selectors.js#L473-L493). It's maybe an abuse of value comparisons in JavaScript, but it works reasonably well and avoids (a) a performance cost associated with populating the full edits / blocks into `AutosaveMonitor` and (b) otherwise exposing internal state structure of the store.

File: packages/editor/src/store/selectors.js [hunk 1]
```diff
@@ -196,6 +196,29 @@ export function getPostEdits( state ) {
...
+ *    getReferenceByDistinctEdits( afterState )
+ * );
+ * ```
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {*} A value whose reference will change only when an edit occurs.
+ */
+export const getReferenceByDistinctEdits = createSelector(
+	() => [],
```
Comment (youknowriad) [inline packages/editor/src/store/selectors.js]: Nice trick :) 

This PR assumes we track `isDirty` by the presence of `edits`, right? Can it create false positives if it's not the case?

**Review by youknowriad — APPROVED** [review]
Works great in my testing

**Pull request merged**